### PR TITLE
✨Enhance accessibility of Stripe Payments Demo template 

### PR DESCRIPTION
**Pull request by barrymcgee**
Assuming developers will use this code as a template, I've made some small tweaks to improve the accessibility of the page overall. 

These changes will bump the Chrome DevTools Accessibility Audit score up from 75 to 96.

The remaining issues are color contrast issues. The soft, muted grey/blue hues do not meet minimum [WCAG 2.0 standards on contrast](https://www.w3.org/TR/WCAG20-TECHS/G18.html) for user with sight limitations. However, as this issue is closely tied with Stripe brand identity, I didn't feel this was an issue I could address.

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -14,8 +16,8 @@
 </head>
 
 <body>
-  <div id="main" class="loading">
-    <header>
+  <main id="main" class="loading">
+    <header class="header">
```
Comment (thorsten-stripe) [inline public/index.html]: Does assigning a header class here have any benefit?

Comment (barrymcgee) [reply public/index.html]: The styling of this element here is shared with the `<h1>` wrapping element in the Order Summary `<aside>` to create the visual illusion of one full width header element. 

This was previously also a `<header>` element but I switched it to a `<div>` and gave them both a `.header` class so they would continue to share the same styling. I switched it to a `<div>` as having them both as header elements mean they both have an implicit `role="banner"` whereas a page should only have one element with a banner role. 

The alternate way is to leave the `<header>` without a class and duplicate the selectors in the CSS like so;

```
.header,
header {
// ...styles
}
```

...but I figure this approach is cleaner.

Comment (thorsten-stripe) [reply public/index.html]: Ah yes, I missed that the aside also had a header element. 
LGTM
Thanks again for submitting this!

Comment (barrymcgee) [reply public/index.html]: Pleasure - loving everything Stripe! 👌🏻

**Pull request merged**